### ISSUE TITLE
Task Dependencies Not Run When Exclusively Using Project-Specific Target Configurations

### ISSUE THREAD

**Issue by rhermes62** [lerna/lerna#3407]
## Current Behavior

In Lerna <=5.5.4, Lerna powered by nx supported exclusively using [project specific target configurations](https://lerna.js.org/docs/api-reference/configuration#project-specific-configuration).

However, in Lerna >=5.6.0 after [this commit](https://github.com/lerna/lerna/commit/51f80d901fceb285677bd55ef2a456f3fb264c13), this support was dropped.

Looked into it and looks like the `excludeTaskDependencies` flag is set [here](https://github.com/lerna/lerna/blob/4fcefff1b7d982c83b6d68d0a84d7e878ea87449/commands/run/index.js#L313) which is determined by whether a `nx.json` file exists and whether `targetDependencies` or `targetDefaults` are defined [here](https://github.com/lerna/lerna/blob/4fcefff1b7d982c83b6d68d0a84d7e878ea87449/commands/run/index.js#L239-L241).

This calculation does not take into consideration [project specific target configurations](https://lerna.js.org/docs/api-reference/configuration#project-specific-configuration).

Because the miscalculation of the `excludeTaskDependencies`, our project task dependencies we had defined suddenly stopped working when migrating to any version of Lerna >=5.6.0 given we do not define `targetDependencies` nor `targetDefaults` which is all thats considered in the setting of `excludeTaskDependencies`.

## Expected Behavior

Task dependencies should be run in the presence of project specific configurations without the need for `targetDependencies` or `targetDefaults`.

The `excludeTaskDependencies` should be set to the pseudocode:

```
hasNxJson && (hasTargetDependencies || hasTargetDefaults || hasProjectSpecificTargets)
```

opposed to the current:

```
hasNxJson && (hasTargetDependencies || hasTargetDefaults)
``` 

which lives [here](https://github.com/lerna/lerna/blob/4fcefff1b7d982c83b6d68d0a84d7e878ea87449/commands/run/index.js#L239-L241).

## Steps to Reproduce

[Repo with instructions on how to reproduce](https://github.com/rhermes62/nx-14.6.0-ignores-package.json-config) (pardon original repo naming, thought it was originally a nx bug but its actually a lerna bug setting `excludeTaskDependencies` incorrectly).

Any repo without `targetDependencies` nor `targetDefaults` in `nx.json` but with nested `package.json`->`nx`->`targets` will reproduce it through.

### Failure Logs / Configuration

Before lerna 5.6.0:

```
Running target test for 4 project(s) and 14 task(s) they depend on
```

After migrating to lerna >=5.6.0

```
Running target test for 4 project(s)
```

<details><summary>lerna.json</summary><p>

```json
{
  "version": "0.0.1",
  "packages": ["packages/*"],
  "rejectCycles": true,
  "stream": true,
  "useWorkspaces": true,
  "useNx": true
}
```
</p></details>

### Environment

```
  System:
    OS: macOS 12.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 16.17.1 - /usr/local/opt/node@16/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/opt/node@16/bin/npm
  Utilities:
    Git: 2.37.1 - /usr/bin/git
  npmPackages:
    lerna: 5.5.4 => 5.5.4 
```

**Comment by JamesHenry** [comment]
@rhermes62 Thanks for the detailed report, we will refine this behaviour to pick up on the nx config in the package.json files appropriately in this case

**Comment by rhermes62** [comment]
Confirmed this was fixed for me in 6.1.0 https://github.com/lerna/lerna/releases/tag/v6.1.0

Thank you!

Closing the issue
